src/rpc_mock.rs

```diff
@@ -0,0 +1,155 @@
+// Implementation of RpcRequestHandler trait for testing Rpc requests without i/o
+
+use crate::rpc_request::{RpcClient, RpcError, RpcRequestHandler};
+use serde_json::{self, Number, Value};
+use solana_sdk::hash::Hash;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::{Keypair, KeypairUtil};
+use solana_sdk::system_transaction::SystemTransaction;
+use solana_sdk::transaction::Transaction;
+use std::error;
+use std::io::{Error, ErrorKind};
+use std::net::SocketAddr;
+
+pub enum MockRpcRequest {
```

Comment:
Instead of duplicating all these fields, would it make sense to define an empty struct with the same name instead? And then in the trait implementation, it would do a match against the original enum.

Reply:
I like this idea, but am not sure what the match against the original enum would look like in practice. Can you steer me toward an example?

Reply:
It'd be the existing `RpcRequest` enum.

Reply:
Oh, duh, I see what you mean. I'll try it!

Reply:
The enum can probably hold onto that `RpcRequest` name. Maybe the wrapper can bring back the name `Rpu`? So then there'd be an `Rpu` and a `MockRpu` both which can process `RpcRequest` messages. Just a thought - not sure if that makes sense in this context.

Reply:
It's nice and short, at least! Just pushed what it would look like. Not too bad.
Can you take a look @garious ? I'd love to get this merged before whatever @mvines is about to do to WalletConfig 😉